Backend/app/dashboard/Lib/src/Storage/Queries/Person.hs

```diff
@@ -11,166 +11,253 @@
-          table @PersonT
-            `innerJoin` table @RoleT
-              `Esq.on` ( \(person :& role) ->
-                           person ^. Person.PersonRoleId ==. role ^. Role.RoleTId
-                       )
-            `leftJoin` merchantAccessAggTable
-              `Esq.on` ( \(person :& _ :& (mbPersonId, _mbMerchantShortIds, _mbMerchantOperatingCityIds)) ->
-                           just (person ^. Person.PersonTId) ==. mbPersonId
-                       )
-      where_ $
-        Esq.whenJust_ (liftA2 (,) mbSearchString mbSearchStrDBHash) (filterBySearchString person)
-          &&. Esq.whenJust_ personId (\defaultPerson -> person ^. PersonId ==. val (getId defaultPerson))
-      orderBy [desc $ person ^. PersonCreatedAt]
-      limit limitVal
-      offset offsetVal
-      return (person, role, mbMerchantShortIds, mbMerchantOperatingCityIds)
+findAllWithLimitOffset mbSearchString mbSearchStrDBHash mbLimit mbOffset personId = do
+  dbConf <- getMasterBeamConfig
+  res <- L.runDB dbConf $
+    L.findRows $
+      B.select $
+        B.limit_ limitVal $
+          B.offset_ offsetVal $
+            B.orderBy_ (\(person, _, _) -> B.desc_ person.createdAt) $
+              -- B.aggregate_ (\(person, role, merchantAccess) -> (B.group_ person.id,B.arrayOf_ (merchantAccess.merchantShortId),B.arrayOf_ (merchantAccess.operatingCity))) $
+
+              -- B.aggregate_ (\(person, role, merchantAccess) -> (B.group_ person.id,B.arrayOf_ (merchantAccess.merchantShortId),B.arrayOf_ (merchantAccess.operatingCity))) $
+              B.filter_'
+                ( \(person, _role, _) ->
+                    ( maybe (B.sqlBool_ $ B.val_ True) (\searchString -> B.sqlBool_ (B.concat_ [person.firstName, person.lastName] `B.like_` B.val_ ("%" <> searchString <> "%"))) mbSearchString
+                        B.||?. maybe (B.sqlBool_ $ B.val_ True) (\searchStrDBHash -> person.mobileNumberHash B.==?. B.val_ searchStrDBHash) mbSearchStrDBHash
+                    )
+                      B.&&?. maybe (B.sqlBool_ $ B.val_ True) (\defaultPerson -> person.id B.==?. B.val_ (getId defaultPerson)) personId
+                )
+                $ do
+                  person <- B.all_ (SBC.person SBC.atlasDB)
+                  role <- B.join_' (SBC.role SBC.atlasDB) (\role -> BeamP.roleId person B.==?. BeamR.id role)
+                  merchantAccess <- B.leftJoin_' (B.all_ $ SBC.merchantAccess SBC.atlasDB) (\merchantAccess -> BeamP.id person B.==?. BeamMA.personId merchantAccess)
+                  pure (person, role, merchantAccess)
+  case res of
+    Right res' -> do
+      let person' = (\(person, _, _) -> person) <$> res'
+          role' = (\(_, role, _) -> role) <$> res'
+          merchantAccess' = (\(_, _, merchantAccess) -> merchantAccess) <$> res'
+      p <- catMaybes <$> mapM fromTType' person'
+      r <- catMaybes <$> mapM fromTType' role'
+      ma <- mapM (maybe (pure Nothing) fromTType') merchantAccess'
+      let finalRes = zip3 p r ma
```

Comment:
this three arrays can contain different number of elements because `catMaybes` used separately to each array, so using  `zip3` here is incorrect
i suggest this code. If one of three computations `MaybeT $ fromTType' `will fail to `Nothing`, then all result will fail 

```
    Right res' -> do
      finalRes <- forM res' $ \(person, role, mbMerchantAccess) -> runMaybeT $ do
        p <- MaybeT $ fromTType' person
        r <- MaybeT $ fromTType' role
        ma <- forM mbMerchantAccess (MaybeT . fromTType')
        pure (p, r, ma)
      pure $ groupByPerson $ catMaybes finalRes
    Left _ -> pure []
```